### DISCUSSION TITLE
Support extracting DATE from TIMESTAMPTZ

### DISCUSSION THREAD

**Discussion by plaflamme** [duckdb/duckdb#6869] [Ideas]
## Current State

Currently, if one has a `TIMESTAMPTZ` column and they wish to extract its date part as a `DATE`, they must resort to one of the following:

* Casting through `VARCHAR`:

```sql
SELECT ts_col::VARCHAR::DATE FROM ts
```

* Use `date_part` and `make_date`:

```sql
SELECT make_date(date_part(['year', 'month', 'day'], ts_col)) FROM ts
```

* probably other ways?

Neither solution is particularly satisfying and going through string formatting / parsing seems error prone.

## Proposition: use allow `DATE` field for `EXTRACT`

It would be more idiomatic to allow extracting a `DATE` component from a `TIMESTAMP` type, e.g.:

```sql
SELECT EXTRACT(DATE FROM ts_col) FROM ts
```

An alternative is to support casting from `TIMESTAMPTZ` to `DATE`

```sql
SELECT ts_col::DATE FROM ts
```

But presumably, there's a reason why this isn't supported already.

**Comment by Mytherin** [comment]
Timestamp -> date casts are supported:

```sql
D create table timestamps(ts TIMESTAMP);
D INSERT INTO timestamps VALUES ('1992-01-01 01:02:03');
D SELECT ts::DATE FROM timestamps;
┌──────────────────┐
│ CAST(ts AS DATE) │
│       date       │
├──────────────────┤
│ 1992-01-01       │
└──────────────────┘
```

Perhaps you mean `TIMESTAMPTZ -> DATE` casts?

**Reply by plaflamme** [reply]
Yes, indeed, the column I was working with was `TIMESTAMPTZ`, my bad. I'll adjust the description.
